From 0fb20783eef43b8f31c6d4f0df7bff60cb461c45 Mon Sep 17 00:00:00 2001
From: San Mehat <san@google.com>
Date: Thu, 30 Jul 2009 08:27:38 -0700
Subject: [PATCH 153/247] Revert "mmc: mmcblk: Add new feature 'CONFIG_MMC_BLOCK_PARANOID_RESUME'"

This reverts commit 5ca694dfd981a371e9b18cdd4a89c002ffaccbc5.

Conflicts:

	drivers/mmc/card/Kconfig
	drivers/mmc/card/block.c

Signed-off-by: San Mehat <san@google.com>
---
 drivers/mmc/card/Kconfig |    7 -------
 drivers/mmc/card/block.c |    4 ----
 drivers/mmc/card/queue.c |   28 ----------------------------
 drivers/mmc/card/queue.h |    3 ---
 4 files changed, 0 insertions(+), 42 deletions(-)

diff --git a/drivers/mmc/card/Kconfig b/drivers/mmc/card/Kconfig
index 176199e..86948f9 100644
--- a/drivers/mmc/card/Kconfig
+++ b/drivers/mmc/card/Kconfig
@@ -32,13 +32,6 @@ config MMC_BLOCK_BOUNCE
 
 	  If unsure, say Y here.
 
-config MMC_BLOCK_PARANOID_RESUME
-	bool "Check card status on resume"
-        depends on MMC_BLOCK
-        default y
-        help
-          Nohelp
-
 config MMC_BLOCK_DEFERRED_RESUME
 	bool "Deferr MMC layer resume until I/O is requested"
 	depends on MMC_BLOCK
diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index 9b8f92b..6867441 100644
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -649,10 +649,6 @@ static int mmc_blk_resume(struct mmc_card *card)
 #ifndef CONFIG_MMC_BLOCK_DEFERRED_RESUME
 		mmc_blk_set_blksize(md, card);
 #endif
-
-#ifdef CONFIG_MMC_BLOCK_PARANOID_RESUME
-		md->queue.check_status = 1;
-#endif
 		mmc_queue_resume(&md->queue);
 	}
 	return 0;
diff --git a/drivers/mmc/card/queue.c b/drivers/mmc/card/queue.c
index 5fac19d..7a72e75 100644
--- a/drivers/mmc/card/queue.c
+++ b/drivers/mmc/card/queue.c
@@ -14,9 +14,7 @@
 #include <linux/freezer.h>
 #include <linux/kthread.h>
 #include <linux/scatterlist.h>
-#include <linux/delay.h>
 
-#include <linux/mmc/mmc.h>
 #include <linux/mmc/card.h>
 #include <linux/mmc/host.h>
 #include "queue.h"
@@ -72,33 +70,7 @@ static int mmc_queue_thread(void *d)
 			continue;
 		}
 		set_current_state(TASK_RUNNING);
-#ifdef CONFIG_MMC_BLOCK_PARANOID_RESUME
-		if (mq->check_status) {
-			struct mmc_command cmd;
 
-			do {
-				int err;
-
-				cmd.opcode = MMC_SEND_STATUS;
-				cmd.arg = mq->card->rca << 16;
-				cmd.flags = MMC_RSP_R1 | MMC_CMD_AC;
-
-				mmc_claim_host(mq->card->host);
-				err = mmc_wait_for_cmd(mq->card->host, &cmd, 5);
-				mmc_release_host(mq->card->host);
-
-				if (err) {
-					printk(KERN_ERR "%s: failed to get status (%d)\n",
-					       __func__, err);
-					msleep(5);
-					continue;
-				}
-				printk(KERN_DEBUG "%s: status 0x%.8x\n", __func__, cmd.resp[0]);
-			} while (!(cmd.resp[0] & R1_READY_FOR_DATA) ||
-				(R1_CURRENT_STATE(cmd.resp[0]) == 7));
-			mq->check_status = 0;
-                }
-#endif
 		mq->issue_fn(mq, req);
 	} while (1);
 	up(&mq->thread_sem);
diff --git a/drivers/mmc/card/queue.h b/drivers/mmc/card/queue.h
index 4b11cf2..64e66e0 100644
--- a/drivers/mmc/card/queue.h
+++ b/drivers/mmc/card/queue.h
@@ -17,9 +17,6 @@ struct mmc_queue {
 	char			*bounce_buf;
 	struct scatterlist	*bounce_sg;
 	unsigned int		bounce_sg_len;
-#ifdef CONFIG_MMC_BLOCK_PARANOID_RESUME
-	int			check_status;
-#endif
 };
 
 extern int mmc_init_queue(struct mmc_queue *, struct mmc_card *, spinlock_t *);
-- 
1.7.0.4

